distributed/client.py

```diff
@@ -2773,7 +2774,10 @@ def collections_to_dsk(collections, *args, **kwargs):
         return collections_to_dsk(collections, *args, **kwargs)
 
 
-Executor = Client
+def Executor(*args, **kwargs):
```

Comment:
Could you make `Executor` a subclass of `Client` that raises the warning in `__init__` like

```python
class Executor(Client):
    warnings.warn(...)
    super().__init__(*args, **kwargs)
```